aten/src/ATen/native/cuda/Activation.cu

```diff
@@ -209,18 +144,14 @@ void log_sigmoid_backward_kernel(TensorIterator& iter) {
 // -----------------------------------
 // prelu forward
 // -----------------------------------
-template <typename scalar_t>
-void prelu_cuda_kernel_share_weights(
-  const Tensor& input,
-  Tensor& result,
-  const scalar_t* weight_data)
-{
-  auto iter = TensorIterator::unary_op(result, input);
-
-  at::native::gpu_kernel(iter,
-    [weight_data] GPU_LAMBDA (scalar_t input_val) {
-        return (input_val > 0) ? input_val : *weight_data * input_val;
-    });
+void launch_prelu_cuda_kernel_share_weights(TensorIteratorBase &iter, const TensorBase &weight) {
+  AT_DISPATCH_FLOATING_TYPES_AND(at::ScalarType::Half, iter.input_dtype(), "prelu_cuda", [&] {
+    const auto *weight_data = weight.data_ptr<scalar_t>();
+    at::native::gpu_kernel(iter,
+        [weight_data] GPU_LAMBDA (scalar_t input_val) {
+          return (input_val > 0) ? input_val : *weight_data * input_val;
+        });
+  });
```

Comment:
ugh, all of the code movement makes the diff unreadable

Reply:
The diff ignoring white space changes makes these `launch_` functions clearer:
https://github.com/pytorch/pytorch/pull/64499/files?diff=unified&w=1

The only substantive change is dispatch now uses `iter.input_dtype()` instead of `input.scalar_type()`. 

Everything else is just moving code around. You can see this by diffing the old `Activation.cu` with `Activation.cpp` and looking at what's added by grepping for lines that start with `>` indicating added

```sh
$ git checkout HEAD~1 -- ./Activation.cu
$ diff Activation.cu Activation.cpp -w | grep "^>"
> #include <ATen/native/cuda/Activation.h>
> namespace at { namespace native {
>   launch_log_sigmoid_forward_kernel(iter);
>     auto iter = TensorIterator::unary_op(result, input);
>     launch_prelu_cuda_kernel_share_weights(iter, weight);
>     launch_prelu_cuda_kernel_multi_weights(result, input, weight);
>     at::TensorIterator iter = TensorIteratorConfig()
>         .add_output(input_grad)
>         .add_output(weight_grad_collector)
>         .add_input(input)
>         .add_input(grad_out)
>         .build();
> 
>     launch_prelu_cuda_backward_kernel_share_weights(iter, weight);
>     launch_prelu_cuda_backward_kernel_multi_weights(
>         input, weight, grad_out, input_grad, weight_grad_collector);
> }}  // namespace at::native
```

All of these are:
- Includes & namespace stuff
- calling `launch_`... functions instead of embedding the kernel definition
- Moving `TensorIterator` construction out of the `_impl` functions, so we can pass them as arguments to `launch_`

And the same test works for `RreluWithNoise.cu`:
```sh
$ diff Activation.cu RreluWithNoise.cu -w | grep "^>"
> namespace at { namespace native {
> }}  // namespace at::native
```